inc/task.class.php

```diff
@@ -592,21 +599,23 @@ function prepareTaskjobs($methods = array(), $tasks_id = false) {
                }
             }
 
-            // If task have not reprepare_if_successful, not reprerare
+            // If task have not reprepare_if_successful, do not reprerare
             // successfull taskjobstate
             if (!$result['task']['reprepare_if_successful']) {
                $jobstates_running = $jobstate->find(
-                  implode(" \n", array(
+                  implode(" \n", [
                      "    `itemtype` = '" . $item_type . "'",
                      "AND `items_id` = ".$item_id,
                      "AND `plugin_fusioninventory_taskjobs_id` = ". $job_id,
-                     "AND `state` = '".
-                        PluginFusioninventoryTaskjobstate::FINISHED."'",
+                     "AND `state` IN (".
```

Comment:
If you have only 1 value, use the `=`, not the `IN` because the `=` if more performant